src/js/plugins/plugin.url.js

```diff
@@ -0,0 +1,169 @@
+  urlTrackIndex0: false,
+});
+
+/** @override */
+BookReader.prototype.setup = (function(super_) {
+  return function(options) {
+    super_.call(this, options);
+
+    this.bookId = options.bookId;
+    this.defaults = options.defaults;
+
+    this.locationPollId = null;
+    this.oldLocationHash = null;
+    this.oldUserHash = null;
+  };
+})(BookReader.prototype.setup);
+
+/** @override */
+BookReader.prototype.init = (function(super_) {
+  return function() {
+
+    if (this.options.enableUrlPlugin) {
+      this.bind(BookReader.eventNames.PostInit, function(e, br) {
+        if (br.options.updateWindowTitle) {
+          document.title = br.shortTitle(50);
+        }
+        if (br.options.urlMode === 'hash') {
+          br.urlStartLocationPolling();
+        }
+      });
+
+      this.bind(BookReader.eventNames.fragmentChange,
+        this.urlUpdateFragment.bind(this)
+      );
+    }
+    super_.call(this);
+  };
+})(BookReader.prototype.init);
+
+/**
+ * Returns a shortened version of the title with the maximum number of characters
+ * @param {string} maximumCharacters
+ */
+BookReader.prototype.shortTitle = function(maximumCharacters) {
+  if (this.bookTitle.length < maximumCharacters) {
+    return this.bookTitle;
+  }
+
+  const title = `${this.bookTitle.substr(0, maximumCharacters - 3)}...`;
+  return title;
+};
+
+/**
+ * Starts polling of window.location to see hash fragment changelet
+ */
+BookReader.prototype.urlStartLocationPolling = function() {
+  this.oldLocationHash = this.urlReadFragment();
+
+  if (this.locationPollId) {
+    clearInterval(this.locationPollID);
+    this.locationPollId = null;
+  }
+
+  const updateHash = () => {
+    const newFragment = this.urlReadFragment();
+    const hasFragmentChange = (newFragment != this.oldLocationHash) && (newFragment != this.oldUserHash);
+
+    if (!hasFragmentChange) { return; }
+
+    const params = this.paramsFromFragment(newFragment);
+
+    const updateParams = () => this.updateFromParams(params);
+
+    this.trigger(BookReader.eventNames.stop);
+    if (this.animating) {
+      // Queue change if animating
+      if (this.autoStop) this.autoStop();
+      this.animationFinishedCallback = updateParams;
+    } else {
+      // update immediately
+      updateParams();
+    }
+    this.oldUserHash = newFragment;
+  }
+
+  this.locationPollId = setInterval(updateHash, 500);
+};
+
+/**
+ * Update URL from the current parameters.
+ * Call this instead of manually using window.location.replace
+ */
+BookReader.prototype.urlUpdateFragment = function() {
+  const allParams = this.paramsFromCurrent();
+  const { urlMode, urlTrackIndex0, urlTrackedParams } = this.options;
+
+  if (!urlTrackIndex0
+      && (typeof(allParams.index) !== 'undefined')
+      && allParams.index === 0) {
+    delete allParams.index;
+    delete allParams.page;
+  }
+
+  const params = urlTrackedParams.reduce((validParams = {}, aParam) => {
```

Comment:
No need for default here.

```suggestion
  const params = urlTrackedParams.reduce((validParams, aParam) => {
```